software/webapp/src/main/java/org/apache/brooklyn/entity/webapp/WebAppServiceMethods.java

```diff
@@ -73,17 +73,26 @@ public static boolean isProtocolEnabled(Entity entity, String protocol) {
 
     public static String inferBrooklynAccessibleRootUrl(Entity entity) {
         if (isProtocolEnabled(entity, "https")) {
-            Integer rawPort = entity.getAttribute(HTTPS_PORT);
-            checkNotNull(rawPort, "HTTPS_PORT sensors not set for %s; is an acceptable port available?", entity);
-            HostAndPort hp = BrooklynAccessUtils.getBrooklynAccessibleAddress(entity, rawPort);
-            return String.format("https://%s:%s/", hp.getHostText(), hp.getPort());
+            return inferBrooklynAccessibleRootUrl(entity, HTTPS_PORT);
         } else if (isProtocolEnabled(entity, "http")) {
-            Integer rawPort = entity.getAttribute(HTTP_PORT);
-            checkNotNull(rawPort, "HTTP_PORT sensors not set for %s; is an acceptable port available?", entity);
-            HostAndPort hp = BrooklynAccessUtils.getBrooklynAccessibleAddress(entity, rawPort);
-            return String.format("http://%s:%s/", hp.getHostText(), hp.getPort());
+            return inferBrooklynAccessibleRootUrl(entity, HTTP_PORT);
         } else {
             throw new IllegalStateException("HTTP and HTTPS protocols not enabled for "+entity+"; enabled protocols are "+getEnabledProtocols(entity));
         }
     }
+
+    public static String inferBrooklynAccessibleRootUrl(Entity entity, AttributeSensor<Integer> sensor) {
```

Comment:
I don't think we need inferBrooklynAccessibleRootUrl(Entity entity, AttributeSensor<Integer> sensor) signature as there are not defined other protocols yet and there is double check once in inferBrooklynAccessibleRootUrl(Entity entity) and once here. A sensor can be set in the first method instead and this would simplify the code.